app/js/FileHandler.js

```diff
@@ -1,235 +1,202 @@
-          return this._getConvertedFileAndCache(
-            bucket,
-            key,
-            convertedKey,
-            opts,
-            callback
-          )
-        }
-      }
-    )
-  },
-
-  _getConvertedFileAndCache(bucket, key, convertedKey, opts, callback) {
-    let convertedFsPath = ''
-    const originalFsPath = ''
-    return async.series(
-      [
-        cb => {
-          return this._convertFile(bucket, key, opts, function(
-            err,
-            fileSystemPath,
-            originalFsPath
-          ) {
-            convertedFsPath = fileSystemPath
-            originalFsPath = originalFsPath
-            return cb(err)
-          })
-        },
-        cb => ImageOptimiser.compressPng(convertedFsPath, cb),
-        cb =>
-          PersistorManager.sendFile(bucket, convertedKey, convertedFsPath, cb)
-      ],
-      function(err) {
-        if (err != null) {
-          LocalFileWriter.deleteFile(convertedFsPath, function() {})
-          LocalFileWriter.deleteFile(originalFsPath, function() {})
-          return callback(err)
-        }
-        // Send back the converted file from the local copy to avoid problems
-        // with the file not being present in S3 yet.  As described in the
-        // documentation below, we have already made a 'HEAD' request in
-        // checkIfFileExists so we only have "eventual consistency" if we try
-        // to stream it from S3 here.  This was a cause of many 403 errors.
-        //
-        // "Amazon S3 provides read-after-write consistency for PUTS of new
-        // objects in your S3 bucket in all regions with one caveat. The
-        // caveat is that if you make a HEAD or GET request to the key name
-        // (to find if the object exists) before creating the object, Amazon
-        // S3 provides eventual consistency for read-after-write.""
-        // https://docs.aws.amazon.com/AmazonS3/latest/dev/Introduction.html#ConsistencyModel
-        return LocalFileWriter.getStream(convertedFsPath, function(
-          err,
-          readStream
-        ) {
-          if (err != null) {
-            return callback(err)
-          }
-          readStream.on('end', function() {
-            logger.log({ convertedFsPath }, 'deleting temporary file')
-            return LocalFileWriter.deleteFile(convertedFsPath, function() {})
-          })
-          return callback(null, readStream)
+  })
+}
+
+function _getConvertedFileAndCache(bucket, key, convertedKey, opts, callback) {
+  let convertedFsPath
+
+  async.series(
+    [
+      cb => {
+        _convertFile(bucket, key, opts, function(err, fileSystemPath) {
+          convertedFsPath = fileSystemPath
+          cb(err)
         })
+      },
+      cb => ImageOptimiser.compressPng(convertedFsPath, cb),
+      cb => PersistorManager.sendFile(bucket, convertedKey, convertedFsPath, cb)
+    ],
+    function(err) {
+      if (err) {
+        LocalFileWriter.deleteFile(convertedFsPath, function() {})
```

Comment:
The original `originalFsPath` was declared `const` and never modified. The callback from `_convertFile` said `originalFsPath = originalFsPath`, which was just assigning the variable to itself, shadowing the original declaration.

But it doesn't matter, because `_convert` deletes the original file anyway. So, this was a no-op which was trying and failing to do something which had already been done elsewhere.

I think the error would have been less noticeable before we decaffed, as it now shows the original as `const`.